src/nvim/api/buffer.c

```diff
@@ -35,7 +35,8 @@
 ///
 /// @param buffer   Buffer handle
 /// @param[out] err Error details, if any
-/// @return Line count
+/// @return Line count, or \`0` if the buffer has been unloaded (see
+///         |nvim_buf_is_loaded()|).
```

Comment:
perhaps this could be formulated to make it clear if `nvim_buf_line_count` is already used, then `nvim_buf_is_loaded` need not also be used. The later is only referenced because shared info of all buffer functions is placed there, ideally this information could be at `|api-buffer|` before any specific function (but the generation script maybe needs to be changed to not overwrite such a section).

Reply:
I've updated the `nvim_buf_is_loaded()` docs to mention that you can also use `nvim_buf_line_count()`. I have also moved the comments on unloaded buffers to a new intro paragraph at the top of the `api-buffer` section.